### ISSUE TITLE
Add Backup / Restore feature

### ISSUE THREAD

**Issue by InnerCato** [Fox2Code/FoxMagiskModuleManager#52]
If you switch a phone or factory reset you have to reinstall all modules manually but if we could backup we could just restore a file and it would install all the backed up files automatically.

**Comment by Fox2Code** [comment]
Yeah, will not be in `0.3.0` but could be an interesting feature, doing things like module-packs.
But please note that it will require to re-download all installed modules to do the packing, as the installed state of module can be incompatible across versions, compared to the zip state of a module.

**Comment by Fox2Code** [comment]
At also it will not work for all modules.

**Comment by InnerCato** [comment]
Ok, but maybe you could put multiple module's zips in a zip file and lsposed would install all of them instead of doing it manually one by one.

**Comment by InnerCato** [comment]
> Ok, but maybe you could put multiple module's zips in a zip file and lsposed would install all of them instead of doing it manually one by one. 

Like put 2 modules in a zip file

**Comment by InnerCato** [comment]
That sound like a good idea!

**Comment by Fox2Code** [comment]
Why did you closed your issue?
You are only supposed to close it if you are no longer interested by that feature.
Please re-open it if you are still interested.

**Comment by InnerCato** [comment]
I am interested in this being added

**Comment by DestroyerXyz** [comment]
> Yeah, will not be in `0.3.0` but could be an interesting feature, doing things like module-packs. But please note that it will require to re-download all installed modules to do the packing, as the installed state of module can be incompatible across versions, compared to the zip state of a module.

You could backup links to all modules (if they have one) to a file and restore using that.

**Comment by DerGoogler** [comment]
> It also will not work for all modules.

Injected modules get automatic restored, Like HuskyDG Bootloopsaver and my own A70 Overlay.

**Comment by DerGoogler** [comment]
> This is a bad idea on many levels - some modules are device specific, install scripts often are very involved, etc.
> 
> At most a _list_ of modules maybe, but again - this sounds like a lot of work outside the scope of FoxMMM for the problems it could cause

Just add option to export modules as links or something.

**Comment by Fox2Code** [comment]
The idea is to backup module list, and restore it via repo download.

**Comment by DerGoogler** [comment]
> The idea is to backup module list, and restore it via repo download.

Downloaded at once and Installed at once? That can be dangerous... because some modules can have conflicts

**Comment by wefalltomorrow** [comment]
This would be such an awesome feature!

**Comment by Fox2Code** [comment]
Even if it's not currently planed and not easy to do doesn't mean it has to be closed.

**Comment by raulvasquez** [comment]
> 

You could add check boxes for the modules to be reinstalled from the backup saved, and if you prefer, restore them in small batches after rebooting and testing. I def see some value in not needing to remember everything installed. It would also be useful to have a way of tracking and saving manually downloaded module links https://github.com/Fox2Code/FoxMagiskModuleManager/issues/282

**Comment by wefalltomorrow** [comment]
Was this completed?
